pkg/module/storage_gcs.go

```diff
@@ -1,235 +0,0 @@
-		if !ok {
-			continue
-		}
-
-		module := Module{
-			Version: version,
-		}
-
-		modules = append(modules, module)
-	}
-	return modules, nil
-}
-
-func (s *GCSStorage) UploadModule(ctx context.Context, namespace, name, provider, version string, body io.Reader) (Module, error) {
-	if namespace == "" {
-		return Module{}, errors.New("namespace not defined")
-	}
-
-	if name == "" {
-		return Module{}, errors.New("name not defined")
-	}
-
-	if provider == "" {
-		return Module{}, errors.New("provider not defined")
-	}
-
-	if version == "" {
-		return Module{}, errors.New("version not defined")
-	}
-
-	key := storagePath(s.bucketPrefix, namespace, name, provider, version, DefaultArchiveFormat)
-	if _, err := s.GetModule(ctx, namespace, name, provider, version); err == nil {
-		return Module{}, errors.Wrap(ErrAlreadyExists, key)
-	}
-
-	wc := s.sc.Bucket(s.bucket).Object(key).NewWriter(ctx)
-	if _, err := io.Copy(wc, body); err != nil {
-		return Module{}, errors.Wrapf(ErrUploadFailed, err.Error())
-	}
-	if err := wc.Close(); err != nil {
-		return Module{}, errors.Wrapf(ErrUploadFailed, err.Error())
-	}
-
-	return s.GetModule(ctx, namespace, name, provider, version)
-}
-
-// GCSStorageOption provides additional options for the GCSStorage.
-type GCSStorageOption func(*GCSStorage)
-
-// WithGCSStorageBucketPrefix configures the s3 storage to work under a given prefix.
-func WithGCSStorageBucketPrefix(prefix string) GCSStorageOption {
-	return func(s *GCSStorage) {
-		s.bucketPrefix = prefix
-	}
-}
-
-// WithGCSArchiveFormat configures the module archive format (zip, tar, tgz, etc.)
-func WithGCSArchiveFormat(archiveFormat string) GCSStorageOption {
-	return func(s *GCSStorage) {
-		s.archiveFormat = archiveFormat
-	}
-}
-
-// WithGCSStorageSignedURL configures the s3 storage to work under a given prefix.
-func WithGCSStorageSignedURL(set bool) GCSStorageOption {
-	return func(s *GCSStorage) {
-		s.signedURL = set
-	}
-}
-
-// WithGCSServiceAccount configures Application Default Credentials (ADC) service account email.
-func WithGCSServiceAccount(sa string) GCSStorageOption {
-	return func(s *GCSStorage) {
-		s.serviceAccount = sa
-	}
-}
-
-// WithGCSServiceAccount configures Application Default Credentials (ADC) service account email.
-func WithGCSSignedUrlExpiry(seconds int64) GCSStorageOption {
-	return func(s *GCSStorage) {
-		s.signedURLExpiry = seconds
-	}
-}
-
-func NewGCSStorage(bucket string, options ...GCSStorageOption) (Storage, error) {
-	ctx := context.Background()
-	client, err := storage.NewClient(ctx)
-	if err != nil {
-		return nil, err
-	}
-	s := &GCSStorage{
-		sc:            client,
-		bucket:        bucket,
-		archiveFormat: DefaultArchiveFormat,
-	}
-
-	for _, option := range options {
-		option(s)
-	}
-
-	return s, nil
-}
-
-func (s *GCSStorage) generateDownloadURL(bucket, key string) (string, error) {
-	return fmt.Sprintf("gcs::https://www.googleapis.com/storage/v1/%s/%s", bucket, key), nil
-}
```

Comment:
I removed the `generateDownloadURL()` method. I don't know what the `gcs::` prefix is used for. In the docs, I could only find the `https://storage.googleapis.com/` which I left in the `gcs.go` in the `storage` module.